src/fabric/src/fabric2_db.erl

```diff
@@ -946,23 +954,144 @@ fold_docs(Db, UserFun, UserAcc0, Options) ->
                 DocOpts = couch_util:get_value(doc_opts, Options, []),
                 OpenOpts = [deleted | DocOpts],
 
-                Row1 = case lists:keyfind(include_docs, 1, Options) of
-                    {include_docs, true} ->
-                        Row0 ++ open_json_doc(Db, DocId, OpenOpts, DocOpts);
```

Comment:
I wonder if we could try a quick and easy way first just by re-using the transaction so going from Db to TxDb